tests/common/test_cgroupstelemetry.py

```diff
@@ -110,24 +116,16 @@ def test_telemetry_polling_with_active_cgroups(self, *args):
 
                         for data_count in range(poll_count, 10):
                             CGroupsTelemetry.poll_all_tracked()
-                            self.assertEqual(CGroupsTelemetry._cgroup_metrics.__len__(), num_extensions)
-                            for cgroup_name, cgroup_metric in CGroupsTelemetry._cgroup_metrics.items():
-                                current_memory_usage, max_memory_levels, current_cpu_usage = cgroup_metric.get_metrics()
-                                self.assertEqual(len(current_memory_usage._data), data_count)
-                                self.assertListEqual(current_memory_usage._data, [current_memory] * data_count)
-                                self.assertEqual(len(max_memory_levels._data), data_count)
-                                self.assertListEqual(max_memory_levels._data, [current_max_memory] * data_count)
-                                self.assertEqual(len(current_cpu_usage._data), data_count)
-                                self.assertListEqual(current_cpu_usage._data, [current_cpu] * data_count)
+                            self.assertEqual(len(CGroupsTelemetry._cgroup_metrics), num_extensions)
```

Comment:
those checks were repeated over and over so capture them on a helper function _assert_cgroup_metrics_equal